astroid/nodes/node_classes.py

```diff
@@ -4216,6 +4216,107 @@ def get_children(self):
         yield from self.finalbody
 
 
+class TryStar(_base_nodes.MultiLineWithElseBlockNode, _base_nodes.Statement):
+    """Class representing an :class:`ast.TryStar` node."""
+
+    _astroid_fields = ("body", "handlers", "orelse", "finalbody")
+    _multi_line_block_fields = ("body", "handlers", "orelse", "finalbody")
+
+    def __init__(
+        self,
+        *,
+        lineno: int | None = None,
+        col_offset: int | None = None,
+        end_lineno: int | None = None,
+        end_col_offset: int | None = None,
+        parent: NodeNG | None = None,
+    ) -> None:
+        """
+        :param lineno: The line that this node appears on in the source code.
+        :param col_offset: The column that this node appears on in the
+            source code.
+        :param parent: The parent node in the syntax tree.
+        :param end_lineno: The last line this node appears on in the source code.
+        :param end_col_offset: The end column this node appears on in the
+            source code. Note: This is after the last symbol.
+        """
+        self.body: list[NodeNG] = []
+        """The contents of the block to catch exceptions from."""
+
+        self.handlers: list[ExceptHandler] = []
+        """The exception handlers."""
+
+        self.orelse: list[NodeNG] = []
+        """The contents of the ``else`` block."""
+
+        self.finalbody: list[NodeNG] = []
+        """The contents of the ``finally`` block."""
+
+        super().__init__(
+            lineno=lineno,
+            col_offset=col_offset,
+            end_lineno=end_lineno,
+            end_col_offset=end_col_offset,
+            parent=parent,
+        )
+
+    def postinit(
+        self,
+        *,
+        body: list[NodeNG] | None = None,
```

Comment:
All of these should never be None:
https://github.com/python/typeshed/blob/9472dc10b98570633e8777accf339dcec0179da4/stdlib/_ast.pyi#L178

Reply:
But why would we bother to make `TryStar` the only exception? Wouldn't we just batch them all up with `with`, etc.?

Reply:
I don't really understand your comment, but I think it's better to be more restrictive where possible since we don't clearly define our public API here. There is just more options to deal with later if we ever want to refactor this.

I do like that these are keyword only!

Reply:
According to the typing `orelse` and `finalbody` should also only be `Statement` and not `NodeNG` I think, but that's another issue 😓

Reply:
I'm just pointing out that the current typing in this PR matches the typing for many other nodes such as `TryExcept`, `With`, `If`, `For`, etc., and doesn't significantly increase technical debt if we're going to tighten those all in the future. (In fact, it would seem weird to me to consciously fix only _one_ at this time.)

Reply:
Hm, I think it would have been easier to tighten them now since we're "investigating" the node anyway and make a small comment about the typing being correct somewhere.
The issue is that right now we can't know for sure that any of the typing in `astroid` is correct. Marking `TryStart` as "reviewed and actually correct" would have been a good first step I think.

Reply:
> @[DanielNoord](https://github.com/DanielNoord) DanielNoord [4 minutes ago](https://github.com/PyCQA/astroid/pull/2028#discussion_r1125745466)
According to the typing orelse and finalbody should also only be Statement and not NodeNG I think, but that's another issue 😓

Interesting. We should decide if we're moving forward with #1867 before delving into that, I would think. And honestly I'd still prefer fixing all affected nodes in one fell swoop. Better release story esp. if we have to tinker with constructors to not allow None, etc., which is more impactful breakage.

Reply:
Fair enough!